src/core/public/application/types.ts

```diff
@@ -285,6 +285,16 @@ export type AppDeepLink = {
   keywords?: string[];
   /** Optional status of the chrome navigation, defaults to `hidden` */
   navLinkStatus?: AppNavLinkStatus;
+  /** An ordinal used to sort nav links relative to one another for display. */
+  order?: number;
+  /** A tooltip shown when hovering over app link. */
+  tooltip?: string;
+  /** A EUI iconType that will be used for the app's icon. This icon takes precendence over the `icon` property. */
+  euiIconType?: string;
+  /** A URL to an image file used as an icon. Used as a fallback if `euiIconType` is not provided. */
+  icon?: string;
+  /** If set to true, the application's route will only be checked against an {@link App.exactRoute | exact match}. Defaults to `false`. */
+  exactRoute?: boolean;
```

Comment:
I feel like we should extract those fields to their own type to avoid the duplication between `App` and `AppDeepLink` and to avoid having them diverge.

Reply:
I can extract these fields in a new type, let's say _AppNavOptions_, and then do `interface App extends AppNavOptions {...}` and `type AppDeepLink = AppNavOptions & {...}`, 
Another option is to keep these fields inside `App` and use them in the `AppDeepLink` via _Pick_. 
What seems better to you?

Reply:
I think using a proper parent type is better, so let's go with `AppNavOptions`